### ISSUE TITLE
Extra top space (an empty paragraph) if used inside tcolorbox

### ISSUE THREAD

**Issue by muzimuzhi** [muzimuzhi/thmtools#16]
From https://github.com/T-F-S/tcolorbox/issues/122. Note that theorems defined by `amsthm` work ok.

The problem occurs when
 - `\parskip` is non-zero and
 - `tcolorbox` option `parbox=false` is set.

```tex
\documentclass{article}
\usepackage{tcolorbox} % reproducible even prior to tcolorbox v4.32
\usepackage{lipsum}

\usepackage{amsthm}
\newtheorem{thm}{Amsthm}
\usepackage{thmtools}
\declaretheorem[name=Thmtools]{theorem}

\parskip=10pt
\def\dummy{
  Nam dui ligula, fringilla a, euismod sodales, sollicitudin vel, wisi. \par
  Morbi auctor lorem non justo. Nam lacus libero, pretium at, lobortis vitae, ultricies et, tellus.}

\begin{document}
\tcbset{parbox=false}

\begin{tcolorbox}
  \begin{theorem} \dummy \end{theorem}
\end{tcolorbox}

\begin{tcolorbox}
  \begin{thm} \dummy \end{thm}
\end{tcolorbox}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/95097428-0eb61180-0760-11eb-8094-4def75c04ddd.png)

**Comment by mbertucci47** [comment]
The output is different now; both cases have the extra space.

**Comment by muzimuzhi** [comment]
Looks like (another) `parbox=false` problem from `tcolorbox`. Not sure if it's intended.

Tested on overleaf.com, `tcolorbox` v4.30 (2020-04-28) won't cause empty paragraph in the `parbox=false` box in your example.

**Comment by mbertucci47** [comment]
I've added a comment to the corresponding closed [tcolorbox issue](https://github.com/T-F-S/tcolorbox/issues/123#issuecomment-1880309128).

**Comment by muzimuzhi** [comment]
I opened T-F-S/tcolorbox#262.

**Comment by mbertucci47** [comment]
Fixed with [tcolorbox #262](https://github.com/T-F-S/tcolorbox/issues/262).
